File: solutions/java/lasagna/3/src/main/java/Lasagna.java
public class Lasagna {

    public static final int OVEN_TIME = 40;
    public static final int TIMER_PER_LAYER = 2;
    
    public int expectedMinutesInOven() {
        return OVEN_TIME;
    }
    
    public int remainingMinutesInOven(int actualMinutesInOven) {
        return expectedMinutesInOven() - actualMinutesInOven;
    }
    
    public int preparationTimeInMinutes(int numberOfLayers) {
        return numberOfLayers * TIMER_PER_LAYER;
    }

    public int totalTimeInMinutes(int numberOfLayers, int actualMinutesInOven) {
        return preparationTimeInMinutes(numberOfLayers) + actualMinutesInOven;
    }
    
}

File: solutions/python/guidos-gorgeous-lasagna/3/lasagna.py
EXPECTED_BAKE_TIME = 40
TIME_COOKING_A_LAYER = 2

def bake_time_remaining(actual_minutes_in_oven):
    """
    """
    return EXPECTED_BAKE_TIME - actual_minutes_in_oven

def preparation_time_in_minutes(number_of_layers):
    """
    """
    return TIME_COOKING_A_LAYER * number_of_layers

def elapsed_time_in_minutes(number_of_layers, elapsed_bake_time):
    """Calculate the elapsed cooking time.

    :param number_of_layers: int - the number of layers in the lasagna.
    :param elapsed_bake_time: int - elapsed cooking time.
    :return: int - total time elapsed (in minutes) preparing and cooking.

    This function takes two integers representing the number of lasagna layers and the
    time already spent baking and calculates the total elapsed minutes spent cooking the
    lasagna.
    """
    return elapsed_bake_time + preparation_time_in_minutes(number_of_layers)
